data/contracts_finder/processing/scripts/cf_notices_ocds_api.py

```diff
@@ -0,0 +1,416 @@
+                self.fromDate = kwargs['fromdate']
+            else:
+                self.fromDate = str(date.today() - timedelta(days=7))
+            if kwargs.get('todate') != None:
+                self.toDate = kwargs['todate']
+            else:
+                self.toDate = str(date.today())
+
+        if self.dataset == 'bodsmatch':
+            self.dataset_dir = 'json_1_1_bods_match'
+        elif self.dataset == 'suppliermatch':
+            self.dataset_dir = 'json_1_1_supplier_ids_match'
+        elif self.dataset == 'raw':
+            self.dataset_dir = 'json_1_1_raw'
+
+    def get_notices(self, datasetpth='json_1_1_bods_match'):
+        """ Get notices from Contracts Finder API
+            With option of searching date periods
+            or searching existing OCIDs that have a sufficient number of matched suppliers.
+         """
+        session = session_with_backoff()
+        if self.saved_files:
+            dirpath = os.path.join(OCDS_OUTPUT_DIR, datasetpth)
+            for filename in os.listdir(dirpath):
+                filepath = os.path.join(dirpath, filename)
+                with open(filepath) as notice:
+                    resp = json.load(notice)
+                    ocds = json.loads(session.get(resp['uri']).content)
+                    yield ocds
+        else:
+
+            notices = search(publishedFrom=self.fromDate, publishedTo=self.toDate)
+            for n in notices:
+                yield n
+
+
+def fix_cf_supplier_ids(json):
+    """ Modify Contracts Finder OCDS JSON enough to be processed """
+    try:
+        # Fixing to match current version, to be processed by ocdskit
+        json['version'] = '1.0'
+        # Giving suppliers distinct IDs (source sets all supplier IDs to zero)
+        for i, supplier in enumerate(json['releases'][0]['awards'][0]['suppliers']):
+            json['releases'][0]['awards'][0]['suppliers'][i]['id'] = str(i)
+    except:
+        logging.info('Failed to add supplier IDs', exc_info=True)
+    return json
+
+
+def clean_and_convert_to_1_1(row):
+    try:
+        rowjson = json.dumps(row)
+        rowjson = json.loads(rowjson)
+
+        # Fix OCDS errors
+        rowjson = fix_cf_supplier_ids(rowjson)
+
+        # Convert to 1.1 OCDS
+        json_1_1 = upgrade_10_11(json.loads(json.dumps(rowjson), object_pairs_hook=OrderedDict))
+        # json_1_1 = json.dumps(json_1_1)
+        return json_1_1
+    except:
+        logging.debug('Failed to get 1.1 OCDS', exc_info=True)
+
+
+def match_supplier_info(suppliername):
+    """ Adding supplier details from table where a match is found. """
+    try:
+        row = supplier_df.loc[suppliername]
+        supplier_match = {
+            'id': row.iloc[0]['id'],
+            'scheme': row.iloc[0]['scheme'],
+            'legalName': suppliername
+        }
+        return supplier_match
+    except Exception as e:
+        logger.debug('Supplier match not found for %s', suppliername)
+        return None
```

Comment:
Could a match not being found be checked for without a try/except? And debug logging might hide this error